src/scipp/io/csv.py

```diff
@@ -0,0 +1,141 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+
+"""Load CSV files.
+
+Note
+----
+CSV support requires `pandas <https://pandas.pydata.org/>`_ to be installed.
+You need to do this manually as it is not declared as a dependency of Scipp.
+You can use either ``pip install pandas`` or ``conda install -c conda-forge pandas``.
+
+
+CSV ('comma separated values') files store a simple table of data as a string.
+There are many different forms of this format.
+So check whether the loaded data is what you expect for your files.
+See ``scipp.io.csv.load_csv`` for examples.
+
+See Also
+--------
+pandas.read_csv:
+    More details on the underlying parser.
+"""
+
+from io import BytesIO, StringIO
+from os import PathLike
+from typing import Iterable, Optional, Union
+
+from ..compat.pandas_compat import HeadParserArg, from_pandas
+from ..core import Dataset
+
+
+# The typehint of filepath_or_buffer is less generic than in pd.read_table
+# because the definitions of protocols are private in pandas.
+def _load_dataframe(
+    filepath_or_buffer: Union[str, PathLike, StringIO, BytesIO], sep: str
+):
+    try:
+        import pandas as pd
+    except ImportError:
+        raise ImportError(
+            "Pandas is required to load CSV files but not install. "
+            "Install it with `pip install pandas` or "
+            "`conda install -c conda-forge pandas`."
+        ) from None
+    return pd.read_table(filepath_or_buffer, sep=sep)
+
+
+def load_csv(
```

Comment:
Why are we limiting ourselves to this particular function? Pandas has many options for reading files (https://pandas.pydata.org/pandas-docs/stable/user_guide/io.html), so why are we singling out this one? Can we have a generic `load_with_pandas(read=pd.read_csv, ...)`, for example? Or do you plan to wrap more functions in the future? Writing all the docstrings may become tedious?

Reply:
Or why not have people use pandas directly, then convert? It seems `load_csv` is trivial. What is wrong with doing it in two steps?

Reply:
> What is wrong with doing it in two steps?

Nothing really. But it's more convenient this way. People don't even have to know about pandas except that it needs to be installed.
Also, this allows us to replace the implementation if it turns out that we don't want to use pandas to implement this.

Reply:
Not sure I fully agree about the convenience? The Pandas readers have tons of options, and in practice many CSV (or other) files may require using that. Also, if users are familiar with Pandas (which is likely), then having them learn the names of a bunch of Scipp wrappers isn't really user friendly.

> Also, this allows us to replace the implementation if it turns out that we don't want to use pandas to implement this.

Well, only as long as we do not support forwarding arguments to Pandas. My guess is that we would sooner or later add that, since otherwise the wrapper is useless for a major fraction of all possible files. At that point you get locked into Pandas. So why not focus on making a really good converter? I do not feel a need to "hide" the fact that we are using Pandas.